### PR TITLE
multiply `GAS_DENSITY` by `densityRatio`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add type of the species to template signature of
  - the kernel `kernelFillGridWithParticles`
  - the function `calcRealDensity `
- multiply `GAS_DENSITY` by `densityRatio` of the species

~~Do not merge before #955~~

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
rebased

**Comment by PeterChin** [comment]
A query: if l want to code Navier-Stokes to test a global solution, would this be of interest to you? Sorry to butt in.

**Comment by ax3l** [comment]
@PeterChin can you please open a new issue for that and describe your question in way more detail? it's very off-topic in this pull request. Thanks :)

**Comment by ax3l** [comment]
@psychocoderHPC thx! nice :)

File: src/picongpu/include/particles/ParticlesInit.kernel [hunk 1]
```diff
@@ -47,17 +48,19 @@ using namespace PMacc;
...
  */
-template<typename T_GasProfile>
+template< typename T_Species, typename T_GasProfile>
 DINLINE float_X calcRealDensity(T_GasProfile& gasFunctor,
                                 const DataSpace<simDim>& totalGpuCellIdx)
 {
     PMACC_CASSERT_MSG(Please_deselect_gasInitMethod_for_your_species_or_set_GAS_DENSITY_to_a_value_greater_than_0, ((GAS_DENSITY) > 0.0f));
 
-    const float_X value = gasFunctor(totalGpuCellIdx) * GAS_DENSITY;
+    const float_X densityRatioOfSpecies = traits::GetDensityRatio<T_Species>::type::getDefaultValue();
```
Comment (ax3l) [inline src/picongpu/include/particles/ParticlesInit.kernel]: bug: should have been `getValue()` :beetle:

**Pull request merged**